integration/tests/access/cohort3/execution_data_pruning_test.go

```diff
@@ -0,0 +1,270 @@
+func (s *ExecutionDataPruningSuite) waitUntilExecutionDataForBlockIndexed(waitingBlockHeight uint64) {
+	observerNode := s.net.ContainerByName(s.observerNodeName)
+
+	grpcClient, err := s.getGRPCClient(observerNode.Addr(testnet.GRPCPort))
+	s.Require().NoError(err)
+
+	// creating execution data api client
+	client, err := getClient(fmt.Sprintf("localhost:%s", observerNode.Port(testnet.ExecutionStatePort)))
+	s.Require().NoError(err)
+
+	// pause until the observer node start indexing blocks
+	s.Require().Eventually(func() bool {
+		_, err := grpcClient.GetEventsForHeightRange(s.ctx, &accessproto.GetEventsForHeightRangeRequest{
+			Type:                 sdk.EventAccountCreated,
+			StartHeight:          1,
+			EndHeight:            1,
+			EventEncodingVersion: entities.EventEncodingVersion_CCF_V0,
+		})
+
+		return err == nil
+	}, 2*time.Minute, 10*time.Second)
+
+	// subscribe on events till waitingBlockHeight to make sure that execution data for block indexed till waitingBlockHeight and pruner
+	// pruned execution data at least once
+	// SubscribeEventsFromStartHeight used as subscription here because we need to make sure that execution data are already indexed
+	stream, err := client.SubscribeEventsFromStartHeight(s.ctx, &executiondata.SubscribeEventsFromStartHeightRequest{
+		EventEncodingVersion: entities.EventEncodingVersion_CCF_V0,
+		Filter:               &executiondata.EventFilter{},
+		HeartbeatInterval:    1,
+		StartBlockHeight:     0,
+	})
+	s.Require().NoError(err)
+	eventsChan, errChan, err := SubscribeHandler(s.ctx, stream.Recv, eventsResponseHandler)
+	s.Require().NoError(err)
+
+	for {
+		select {
+		case err := <-errChan:
+			s.Require().NoErrorf(err, "unexpected %s error", s.observerNodeName)
+		case event := <-eventsChan:
+			if event.Height > waitingBlockHeight {
+				return
+			}
+		}
+	}
+}
+
+// checkResults checks the results of execution data pruning to ensure correctness.
+func (s *ExecutionDataPruningSuite) checkResults(
+	headers *storage.Headers,
+	anResults *storage.ExecutionResults,
+	onResults *storage.ExecutionResults,
+	anEds execution_data.ExecutionDataStore,
+	onEds execution_data.ExecutionDataStore,
+) {
+	// Loop through blocks and verify the execution data was pruned correctly
+	// execution data till height equal threshold + 1 should be pruned
+
+	// checking execution results from 1 block
+	startBlockHeight := uint64(1)
+	for i := startBlockHeight; i <= s.threshold+1; i++ {
+		header, err := headers.ByHeight(i)
+		require.NoError(s.T(), err, "%s: could not get header", s.accessNodeName)
+
+		result, err := anResults.ByBlockID(header.ID())
+		require.NoError(s.T(), err, "%s: could not get sealed result", s.accessNodeName)
+
+		// verify AN execution data
+		_, err = anEds.Get(s.ctx, result.ExecutionDataID)
+		require.Error(s.T(), err, "%s: could not prune execution data for height %v", s.accessNodeName, i)
+		require.ErrorContains(s.T(), err, "not found")
```

Comment:
check against the sentinel whenever possible
```suggestion
		require.ErrorIs(s.T(), err, storage.ErrNotFound)
```

Reply:
[0cba0a9](https://github.com/onflow/flow-go/pull/6217/commits/0cba0a9ff7fa03f438e75efc5b93223ad8fc3089)